src/providers/aws.rs

```diff
@@ -220,14 +198,27 @@ where
         self
     }
 
-    // TODO with rust specialization (https://github.com/rust-lang/rust/issues/31844), can do:
-    // ```
-    // self.credential_provider.unwrap_or_else(|| Box::new(|| DefaultCredentialProvider::new()))?()
-    // ```
+    /// A closure which returns [`P:
+    /// ProvideAwsCredentials`](https://docs.rs/rusoto_core/0.40.0/rusoto_core/trait.ProvideAwsCredentials.html).
```

Comment:
This comment doesn't seem quite right. And could also probably be more helpful :p